# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/interfaces/security/nsIContentSecurityPolicy.idl
# Commit: 0785d6bfebb3
# Full Hash: 0785d6bfebb3405227469a255389dac80b627f91
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2025-01-17 21:57:54
# Regressor Bug: 1901492
# File Overlap Count: 1
# Description:
#   Bug 1901492 - Introduce RequireTrustedTypesForDirectiveState enum for CSP. r=smaug
#   
#   Extend the hasPolicyWithRequireTrustedTypesForDirective boolean on
#   nsIContentSecurityPolicy to an enum in order to indicate whether there
#   is actually a require-trusted-types-for directive with "enforce"
# ==============================================================================

diff -r 23ca1773d73a -r 0785d6bfebb3 dom/interfaces/security/nsIContentSecurityPolicy.idl
--- a/dom/interfaces/security/nsIContentSecurityPolicy.idl	Fri Jan 17 13:35:39 2025 +0000
+++ b/dom/interfaces/security/nsIContentSecurityPolicy.idl	Fri Jan 17 13:35:39 2025 +0000
@@ -75,6 +75,12 @@
     REPORT_TO_DIRECTIVE                   = 27,
   };
 
+  cenum RequireTrustedTypesForDirectiveState : 8 {
+    NONE         = 0,
+    REPORT_ONLY  = 1,
+    ENFORCE      = 2,
+  };
+
   /**
    * Accessor method for a read-only string version of the policy at a given
    * index.
@@ -131,9 +137,14 @@
                     in boolean deliveredViaMetaTag);
 
   /**
-   * Whether a policy with a RequireTrustedTypes was appended
+   * Indicate the global state for the require-trusted-types-for directives
+   * - NONE: No require-trusted-types-for directives.
+   * - REPORT_ONLY: At least one require-trusted-types-for directive, but all
+   *   of them have "report" disposition.
+   * - ENFORCE: At least one require-trusted-types-for directive with "enforce"
+   *   disposition.
    */
-  [infallible] readonly attribute boolean hasPolicyWithRequireTrustedTypesForDirective;
+  [infallible] readonly attribute nsIContentSecurityPolicy_RequireTrustedTypesForDirectiveState requireTrustedTypesForDirectiveState;
 
   /*
    * Whether this policy allows inline script or style.
@@ -380,6 +391,7 @@
 };
 
 typedef nsIContentSecurityPolicy_CSPDirective CSPDirective;
+typedef nsIContentSecurityPolicy_RequireTrustedTypesForDirectiveState RequireTrustedTypesForDirectiveState;
 
 /* Listener for security policy violation event */
 [function, scriptable, uuid(c3163b14-3a8f-46dd-a4af-bd04680364cd)]